internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -451,6 +451,28 @@ func flattenProcessArgs(p *admin20231115.ClusterDescriptionProcessArgs) []map[st
 
 func FlattenAdvancedReplicationSpecsOldSDK(ctx context.Context, apiObjects []admin20231115.ReplicationSpec, rootDiskSizeGB float64, tfMapObjects []any,
```

Comment:
do you think we can intensify unit tests around all this logic?

Reply:
FlattenAdvancedReplicationSpecsOldSDK does have unit testing, will create ticket so we include for FlattenAdvancedReplicationSpecs as well (prefer to handle in separate PR).